### PR TITLE
Feat(i18n): introduce

### PR DESCRIPTION
**Pull request by AugustinMauroy**
# WIP

### CODE REVIEW

**Comment by AugustinMauroy** [comment]
@anonrig what do you think about direction I have take ?

File: components/main-nav.tsx [hunk 1]
```diff
@@ -15,20 +15,22 @@ export function MainNav() {
         <span className='hidden font-bold sm:inline-block'>Ada URL</span>
       </Link>
       <nav className='flex items-center space-x-6 text-sm font-medium'>
+        {/*@TODO: add localized link*/}
         <Link
-          href='/docs'
+          href='en/docs/introduction'
```
Comment (anonrig) [inline components/main-nav.tsx]: Link component receives a locale property

File: app/[lang]/docs/[[...slug]]/page.tsx [hunk 2]
```diff
@@ -56,15 +60,14 @@ export async function generateMetadata({ params }: DocPageProps): Promise<Metada
   }
 }
 
-export async function generateStaticParams(): Promise<DocPageProps['params'][]> {
+export async function generateStaticParams(): Promise<{ params: DocPageProps['params'] }[]> {
```
Comment (anonrig) [inline app/[lang]/docs/[[...slug]]/page.tsx]: This is wrong

**Comment by anonrig** [comment]
I think we should use cookie to store locale and use a middleware to redirect. As well as next link supports locales so, you don't have to have /en/ for each path.

**Comment by anonrig** [comment]
The folder structure of content should be the same as the route path. Such as content/en/docs/performance should load localhost:3000/en/docs/performance etc. And I think we should use fallback and render english version of a document which is not translated.

File: app/[lang]/docs/[[...slug]]/page.tsx (hunk 2 above)
Comment (AugustinMauroy) [reply app/[lang]/docs/[[...slug]]/page.tsx]: Fixed

File: components/main-nav.tsx (hunk 1 above)
Comment (AugustinMauroy) [reply components/main-nav.tsx]: For moment I use `LocalizedLink` because middleware isn't working

**Comment by AugustinMauroy** [comment]
La pr is "out of date". So I'm closing it, if anyone wants to take the initiative again, it's up to them.

**Pull request closed without merging**